looks like it's fixed, just have to think a bit how to handle the Warnings (deprecation appears when re-reading pickle , guess we'll have to update the pickle version ? 

> FAILED signal/tests/test_spectral_estimation.py::TestPsd::test_ppsd_w_iris_against_obspy_results - numpy.exceptions.VisibleDeprecationWarning: dtype(): align should be passed as Python or NumPy boolean but got `align=0`. Did you mean to pass a tuple to create a subarray type? (Deprecated NumPy 2.4)
> FAILED signal/tests/test_spectral_estimation.py::TestPsd::test_ppsd_temporal_plot - numpy.exceptions.VisibleDeprecationWarning: dtype(): align should be passed as Python or NumPy boolean but got `align=0`. Did you mean to pass a tuple to create a subarray type? (Deprecated NumPy 2.4)
> FAILED signal/tests/test_spectral_estimation.py::TestPsd::test_ppsd_spectrogram_plot - numpy.exceptions.VisibleDeprecationWarning: dtype(): align should be passed as Python or NumPy boolean but got `align=0`. Did you mean to pass a tuple to create a subarray type? (Deprecated NumPy 2.4)
> FAILED signal/tests/test_spectral_estimation.py::TestPsd::test_exception_reading_newer_npz - numpy.exceptions.VisibleDeprecationWarning: dtype(): align should be passed as Python or NumPy boolean but got `align=0`. Did you mean to pass a tuple to create a subarray type? (Deprecated NumPy 2.4)
> FAILED signal/tests/test_spectral_estimation.py::TestPsd::test_can_read_npz_without_pickle - numpy.exceptions.VisibleDeprecationWarning: dtype(): align should be passed as Python or NumPy boolean but got `align=0`. Did you mean to pass a tuple to create a subarray type? (Deprecated NumPy 2.4)

--- a/obspy/signal/tests/test_spectral_estimation.py
+++ b/obspy/signal/tests/test_spectral_estimation.py
@@ -328,12 +328,15 @@ class TestPsd:
         inv = read_inventory(testdata['IUANMO.xml'])
 
         # load expected results, for both only PAZ and full response
-        filename_paz = testdata['IUANMO_ppsd_paz.npz']
-        results_paz = PPSD.load_npz(filename_paz, metadata=None,
-                                    allow_pickle=True)
-        filename_full = testdata['IUANMO_ppsd_fullresponse.npz']
-        results_full = PPSD.load_npz(filename_full, metadata=None,
-                                     allow_pickle=True)
+        # we need to catch numpy 2.4.0 warnings (#3668)
+        with warnings.catch_warnings(record=True):
+            warnings.simplefilter('always')
+            filename_paz = testdata['IUANMO_ppsd_paz.npz']
+            results_paz = PPSD.load_npz(filename_paz, metadata=None,
+                                        allow_pickle=True)
+            filename_full = testdata['IUANMO_ppsd_fullresponse.npz']
+            results_full = PPSD.load_npz(filename_full, metadata=None,
+                                         allow_pickle=True)
 
         # Calculate the PPSDs and test against expected results
         # first: only PAZ
@@ -823,7 +826,10 @@ class TestPsd:
             data = np.load(testdata['ppsd_kw1_ehz.npz'], allow_pickle=True)
         # we have to load, modify 'ppsd_version' and save the npz file for the
         # test..
-        items = {key: data[key] for key in data.files}
+            # we need to catch numpy 2.4.0 warnings (#3668)
+        with warnings.catch_warnings(record=True):
+            warnings.simplefilter('always')
+            items = {key: data[key] for key in data.files}
         # deliberately set a higher ppsd_version number
         items['ppsd_version'] = items['ppsd_version'].copy()
         items['ppsd_version'].fill(100)

--- a/obspy/signal/tests/test_spectral_estimation.py
+++ b/obspy/signal/tests/test_spectral_estimation.py
@@ -724,7 +724,6 @@ class TestPsd:
             ppsd = PPSD.load_npz(testdata['ppsd_kw1_ehz.npz'],
                                  allow_pickle=True)
 
-
         restrictions = {'starttime': UTCDateTime(2011, 2, 6, 1, 1),
                         'endtime': UTCDateTime(2011, 2, 7, 21, 12),
                         'year': [2011],

--- a/obspy/signal/tests/test_spectral_estimation.py
+++ b/obspy/signal/tests/test_spectral_estimation.py
@@ -718,7 +718,7 @@ class TestPsd:
         """
         Test plot of several period bins over time
         """
-        # we need to catch numpy 2.4.0 warnings (#3667)
+        # we need to catch numpy 2.4.0 warnings (#3668)
         with warnings.catch_warnings(record=True):
             warnings.simplefilter('always')
             ppsd = PPSD.load_npz(testdata['ppsd_kw1_ehz.npz'],
@@ -761,7 +761,7 @@ class TestPsd:
         """
         Test spectrogram type plot of PPSD
         """
-        # we need to catch numpy 2.4.0 warnings (#3667)
+        # we need to catch numpy 2.4.0 warnings (#3668)
         with warnings.catch_warnings(record=True):
             warnings.simplefilter('always')
             ppsd = PPSD.load_npz(testdata['ppsd_kw1_ehz.npz'],
@@ -816,7 +816,7 @@ class TestPsd:
                "ObsPy version (current 'ppsd_version' is {!s}). Please "
                "consider updating your ObsPy installation.".format(
                    PPSD(stats=Stats(), metadata=None).ppsd_version))
-        # we need to catch numpy 2.4.0 warnings (#3667)
+        # we need to catch numpy 2.4.0 warnings (#3668)
         with warnings.catch_warnings(record=True):
             warnings.simplefilter('always')
             # 1 - loading a npz
@@ -865,7 +865,7 @@ class TestPsd:
         allow np.load the use of pickle, or that a helpful error message is
         raised if allow_pickle is required. See #2409.
         """
-        # we need to catch numpy 2.4.0 warnings (#3667)
+        # we need to catch numpy 2.4.0 warnings (#3668)
         with warnings.catch_warnings(record=True):
             warnings.simplefilter('always')
             # Init a test PPSD and empty byte stream.
@@ -906,7 +906,7 @@ class TestPsd:
             temp_path = ntemp.name
             self._save_npz_require_pickle(temp_path, ppsd)
             # We should be able to load the files when allowing pickle.
-            # we need to catch numpy 2.4.0 warnings (#3667)
+            # we need to catch numpy 2.4.0 warnings (#3668)
             with warnings.catch_warnings(record=True):
                 warnings.simplefilter('always')
                 ppsd.add_npz(temp_path, allow_pickle=True)

